lib/bolt/target.rb

```diff
@@ -157,9 +157,15 @@ class Target
 
     # Satisfies the Puppet datatypes API
     def self.from_asserted_hash(hash)
+      Puppet.lookup(:bolt_inventory).target_deprecation if hash['uri'] && hash['options']
```

Comment:
I think this actually needs to raise a different error message. This is raised when
```
Target.new({ 'uri' => 'foo', 'options' => { 'port' => '22' } })
```
Which is still not going to be valid in Bolt 2.0 because we don't support the 'options' key anymore. This needs a different message saying that the 'options' hash key has been deprecated in favor of 'config'.